### PR TITLE
GEODE-5919: Fix documentation bugs

### PR DESCRIPTION
**Pull request by mmartell**
- adding missing class comments
- fixed class comments that were in the wrong place

### CODE REVIEW

File: cppcache/include/geode/PdxSerializable.hpp [hunk 1]
```diff
@@ -34,6 +34,9 @@ namespace client {
 class PdxReader;
 class PdxWriter;
 
+/**
+ * An interface for objects whose state can be written/read as PDX.
```
Comment (igodwin) [inline cppcache/include/geode/PdxSerializable.hpp]: Looks like this was copied form DataSerializable before it had been changed.

"An interface for objects whose contents can be serialized as primitive types."

File: cppcache/include/geode/TypeRegistry.hpp [hunk 2]
```diff
@@ -32,6 +32,11 @@ namespace client {
 class CacheImpl;
 class PdxSerializer;
 
+/**
+ * Registry for custom serializable types, both PDXSerializable and
+ * DataSerializable.
+ *
```
Comment (igodwin) [inline cppcache/include/geode/TypeRegistry.hpp]: Remove extra line.

File: cppcache/include/geode/CacheTransactionManager.hpp [hunk 3]
```diff
@@ -32,6 +32,11 @@ namespace apache {
 namespace geode {
 namespace client {
 
+/**
+ * An interface for objects who perform transaction management
+ *
```
Comment (igodwin) [inline cppcache/include/geode/CacheTransactionManager.hpp]: Remove extra line.

File: clicache/src/IDataSerializable.hpp [hunk 4]
```diff
@@ -37,6 +37,9 @@ namespace Apache
       ref class DataOutput;
       ref class DataInput;
 
+      /// <summary>
+	  /// An interface for objects whose contents can be serialized as primitive types.
```
[the remark this answers is not in the document]
Comment (mmartell) [reply clicache/src/IDataSerializable.hpp]: Good catch. Fixed.

**Pull request merged**